### ISSUE TITLE
Section 4.3.2 of the book contains typo?

### ISSUE THREAD

**Issue by baderstine** [mlr-org/mlr3book#863]
[Section 4.3.2](https://mlr3book.mlr-org.com/chapters/chapter4/hyperparameter_optimization.html#sec-resample-overfitting) contains some confusing content: 

In the following block, all of the reported values are exactly the same (0.09).  
```
round(c(true_generalization = as.numeric(generalization),
  without_nested_resampling = as.numeric(insample),
  with_nested_resampling = as.numeric(outsample)), 2)

      true_generalization without_nested_resampling 
                     0.09                      0.09 
   with_nested_resampling 
                     0.09

```

However, the following text: "We find that the performance estimate from unnested tuning optimistically overestimates the true performance (which could indicate ‘meta-overfitting’ to the specific inner holdout-splits), while the outer estimate from nested resampling works much better." suggests that they should be different.

I tried running this code on my machine and found the following values (rounded to 4 decimal places):
```
      true_generalization without_nested_resampling    with_nested_resampling 
                   0.2925                    0.0606                    0.2900 
```

I would suggest that (1) the values be updated, and the text changed slightly for clarity: 
"We find that the performance estimate from unnested tuning optimistically overestimates the true performance (i.e., it underestimates error, which could indicate ‘meta-overfitting’ to the specific inner holdout-splits), while the outer estimate from nested resampling is a closer estimate of the true error."

**Comment by larskotthoff** [comment]
Thanks for your report -- I'm also getting different numbers, more in line with yours. The output is generated automatically every time the book is rendered. It looks like there's an issue on the github runner that renders the book. I'll investigate.

**Comment by larskotthoff** [comment]
This should be fixed as soon as the new version is deployed -- setting a seed now for a more illustrative result.
